rules/CodeQuality/Rector/LogicalAnd/LogicalToBooleanRector.php

```diff
@@ -55,10 +55,31 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
+        /** @var LogicalAnd|LogicalOr $left */
+        $left = $node->left;
+        if ($this->shouldRecursivelyCall($left)) {
+            /** @var BooleanAnd|BooleanOr $left */
+            $left = $this->refactor($left);
+        }
+
+        /** @var LogicalAnd|LogicalOr $right */
+        $right = $node->right;
+        if ($this->shouldRecursivelyCall($right)) {
+            /** @var BooleanAnd|BooleanOr $right */
+            $right = $this->refactor($right);
+        }
```

Comment:
Recurse call should not be needed, as Rector rule works untill there is a code to be fixed. It's recursive by nature.

Reply:
The original problem stems from the "infinite loop detection" in `\Rector\Core\Validation\InfiniteLoopValidator::process`.

The test fixtures multi_or and multi_and will run into an InfiniteLoopTraversingException.

If I use print_node on the offending node(s) this is the output:
`'(1 === $val1 or 1 === $val2) || 1 === $val3'`
and 
`'(1 === $val1 and 1 === $val2) && 1 === $val3'`

Not sure if this is now a bug in the InfiniteLoopValidator or if it is intended and my recursion is approach is the correct one in this case